File: src/main/java/wraith/fabricaeexnihilo/json/ingredient/FluidIngredientJson.java
package wraith.fabricaeexnihilo.json.ingredient;

import com.google.gson.JsonDeserializationContext;
import com.google.gson.JsonElement;
import com.google.gson.JsonParseException;
import com.google.gson.JsonSerializationContext;
import wraith.fabricaeexnihilo.api.crafting.FluidIngredient;
import wraith.fabricaeexnihilo.json.BaseJson;

import java.lang.reflect.Type;

public final class FluidIngredientJson extends BaseJson<FluidIngredient> {

    private FluidIngredientJson() {}

    public static final FluidIngredientJson INSTANCE = new FluidIngredientJson();

    @Override
    public FluidIngredient deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context) throws JsonParseException {
        return FluidIngredient.fromJson(json, context);
    }


    @Override
    public JsonElement serialize(FluidIngredient src, Type typeOfSrc, JsonSerializationContext context) {
        return src.toJson(context);
    }

}

File: src/main/kotlin/exnihilofabrico/json/FluidIngredientJson.kt
package exnihilofabrico.json

import com.google.gson.*
import com.google.gson.reflect.TypeToken
import exnihilofabrico.api.crafting.FluidIngredient
import java.lang.reflect.Type

object FluidIngredientJson: JsonDeserializer<FluidIngredient>, JsonSerializer<FluidIngredient> {
    val TYPE_TOKEN = object: TypeToken<FluidIngredient>(){}.type
    @Throws(JsonParseException::class)
    override fun deserialize(json: JsonElement, typeOfT: Type, context: JsonDeserializationContext) =
        FluidIngredient.fromJson(json, context)

    @Throws(JsonParseException::class)
    override fun serialize(src: FluidIngredient, typeOfSrc: Type, context: JsonSerializationContext) = src.toJson(context)
}